src/f32-avgpool/gen/f32-avgpool-9p8x-minmax-rvv-c1v.c

```diff
@@ -101,66 +64,31 @@ void xnn_f32_avgpool_minmax_ukernel_9p8x__rvv_c1v(
         vfloat32m1_t sum01678_f32v = __riscv_vfadd_vv_f32m1(sum018_f32v, sum67_f32v, n);
         vfloat32m1_t sum_f32v = __riscv_vfadd_vv_f32m1(sum2345_f32v, sum01678_f32v, n);
         __riscv_vse32_v_f32m1(b, sum_f32v, n); b += n;
-      } while (c != 0);
+      }
     }
 
     size_t k = kernel_elements;
     for (k -= 9; k > 8; k -= 8) {
-      const float* i0 = *input++;
-      assert(i0 != NULL);
-      if XNN_UNPREDICTABLE(i0 != zero) {
-        i0 = (const float*) ((uintptr_t) i0 + input_offset);
-      }
-      const float* i1 = *input++;
-      assert(i1 != NULL);
-      if XNN_UNPREDICTABLE(i1 != zero) {
-        i1 = (const float*) ((uintptr_t) i1 + input_offset);
-      }
-      const float* i2 = *input++;
-      assert(i2 != NULL);
-      if XNN_UNPREDICTABLE(i2 != zero) {
-        i2 = (const float*) ((uintptr_t) i2 + input_offset);
-      }
-      const float* i3 = *input++;
-      assert(i3 != NULL);
-      if XNN_UNPREDICTABLE(i3 != zero) {
-        i3 = (const float*) ((uintptr_t) i3 + input_offset);
-      }
-      const float* i4 = *input++;
-      assert(i4 != NULL);
-      if XNN_UNPREDICTABLE(i4 != zero) {
-        i4 = (const float*) ((uintptr_t) i4 + input_offset);
-      }
-      const float* i5 = *input++;
-      assert(i5 != NULL);
-      if XNN_UNPREDICTABLE(i5 != zero) {
-        i5 = (const float*) ((uintptr_t) i5 + input_offset);
-      }
-      const float* i6 = *input++;
-      assert(i6 != NULL);
-      if XNN_UNPREDICTABLE(i6 != zero) {
-        i6 = (const float*) ((uintptr_t) i6 + input_offset);
-      }
-      const float* i7 = *input++;
-      assert(i7 != NULL);
-      if XNN_UNPREDICTABLE(i7 != zero) {
-        i7 = (const float*) ((uintptr_t) i7 + input_offset);
+      const float *i[8];
+      for (size_t kk = 0; kk < k; ++kk) {
```

Comment:
You need the bounds to be 8, rather than `k`, which is guaranteed to be greater than 8
```suggestion
      for (size_t kk = 0; kk < 8; ++kk) {
```

Reply:
Thanks. Fixed it.